gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java

```diff
@@ -127,6 +130,21 @@ public void write(GrpcLogRecord logProto) {
     }
   }
 
+  @VisibleForTesting
+  static Map<String, String> getCustomTags(Map<String, String> customTags,
+      Map<String, String> locationTags, String destinationProjectId) {
+    Map<String, String> tags = new HashMap<>();
+    String sourceProjectId = locationTags.get("project_id");
+    if (!Objects.equals(sourceProjectId, destinationProjectId)
+        && !Strings.isNullOrEmpty(destinationProjectId)) {
```

Comment:
Shouldn't this ideally be in the reverse order

```
!Strings.isNullOrEmpty(destinationProjectId)
&& !Objects.equals(sourceProjectId, destinationProjectId)
```

Only if `destinationProjectId` is non-null/empty then you bother comparing with `sourceProjectId`

Reply:
Updated.